pkg/gardenlet/controller/shoot/care/health_test.go

```diff
@@ -422,6 +422,274 @@ var _ = Describe("health check", func() {
+					Spec:       machinev1alpha1.MachineDeploymentSpec{Replicas: int32(1)},
+				}
+				machineDeploymentList := &machinev1alpha1.MachineDeploymentList{Items: []machinev1alpha1.MachineDeployment{machineDeployment}}
+				msg, err := CheckNodesScaling(ctx, fakeClient, nodeList, machineDeploymentList, controlPlaneNamespace)
+				Expect(msg).To(Equal(""))
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("should return an error if the machine for a cordoned node is not found", func() {
+				nodeList := &corev1.NodeList{
+					Items: []corev1.Node{
+						{Spec: corev1.NodeSpec{Unschedulable: true}},
+					},
+				}
+				machineDeploymentList := &machinev1alpha1.MachineDeploymentList{Items: []machinev1alpha1.MachineDeployment{}}
+				msg, err := CheckNodesScaling(ctx, fakeClient, nodeList, machineDeploymentList, controlPlaneNamespace)
+				Expect(msg).To(Equal("NodesScalingDown"))
+				Expect(err).To(MatchError(ContainSubstring("machine object for cordoned node \"\" not found")))
+			})
+
+			It("should return an error if the machine for a cordoned node is not deleted", func() {
+				var (
+					nodeName = "foo"
+
+					machineList = &machinev1alpha1.MachineList{
+						Items: []machinev1alpha1.Machine{
+							{ObjectMeta: metav1.ObjectMeta{GenerateName: "obj-", Namespace: controlPlaneNamespace, Labels: map[string]string{"node": nodeName}}},
+						},
+					}
+					nodeList = &corev1.NodeList{
+						Items: []corev1.Node{
+							{
+								ObjectMeta: metav1.ObjectMeta{Name: nodeName},
+								Spec:       corev1.NodeSpec{Unschedulable: true},
+							},
+						},
+					}
+				)
+				for _, machine := range machineList.Items {
+					Expect(fakeClient.Create(ctx, &machine)).To(Succeed())
+				}
+				msg, err := CheckNodesScaling(ctx, fakeClient, nodeList, &machinev1alpha1.MachineDeploymentList{}, controlPlaneNamespace)
+				Expect(msg).To(Equal("NodesScalingDown"))
+				Expect(err).To(MatchError(ContainSubstring("found but corresponding machine object does not have a deletion timestamp")))
+			})
+
+			It("should return an error if there are more nodes then machines", func() {
+				nodeList := &corev1.NodeList{Items: []corev1.Node{{}}}
+				msg, err := CheckNodesScaling(ctx, fakeClient, nodeList, &machinev1alpha1.MachineDeploymentList{}, controlPlaneNamespace)
+				Expect(msg).To(Equal("NodesScalingDown"))
+				Expect(err).To(MatchError(ContainSubstring("too many worker nodes are registered. Exceeding maximum desired machine count")))
+			})
+
+			It("should return progressing for a regular scale down", func() {
+				var (
+					nodeName = "foo"
+
+					machineList = &machinev1alpha1.MachineList{
+						Items: []machinev1alpha1.Machine{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:         "foo",
+									GenerateName: "obj-",
+									Namespace:    controlPlaneNamespace,
+									Labels:       map[string]string{"node": nodeName},
+									Finalizers:   []string{"foo"},
```

Comment:
A little bit more clearer. In multiple places:
```suggestion
									Finalizers:   []string{"in-deletion"},
```